### PR TITLE
feat: restaurant menu

### PR DESCRIPTION
**Pull request by devteapot**

### CODE REVIEW

File: whizzer/features/menu/components/CategoriesSlider/CategoriesSlider.tsx [hunk 1]
```diff
@@ -1,14 +1,74 @@
...
-    <View>
-      <Text>CategoriesSlider</Text>
-    </View>
+    <FlatList 
+      horizontal
+      data={categories}
+      renderItem={({ item, index }) => {
+        const isSelected = item === value;
+
+        return (
```
Comment (LRNZ09) [inline whizzer/features/menu/components/CategoriesSlider/CategoriesSlider.tsx:26]: Make a dedicated component for the item

File: whizzer/features/menu/components/CategoriesSlider/CategoriesSlider.tsx [hunk 2]
```diff
@@ -1,14 +1,74 @@
...
 
-const CategoriesSlider = (props: Props) => {
   return (
-    <View>
-      <Text>CategoriesSlider</Text>
-    </View>
+    <FlatList 
+      horizontal
+      data={categories}
+      renderItem={({ item, index }) => {
```
Comment (LRNZ09) [inline whizzer/features/menu/components/CategoriesSlider/CategoriesSlider.tsx:23]: Make a named function for the render item callback

File: whizzer/features/menu/screens/DishDetailScreen/DishDetailScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { StatusBar } from 'expo-status-bar';
+import React from 'react';
+import { StyleSheet, View, Text } from 'react-native';
+import { RootScreenNames } from '../../../../navigation/constants';
+import { RootStackScreenProps } from '../../../../types';
+
+type DishDetailScreenProps = RootStackScreenProps<RootScreenNames.DishDetailScreen>;
+
+const DishDetailScreen: React.FC<DishDetailScreenProps> = ({ route }) => {
```
Comment (LRNZ09) [inline whizzer/features/menu/screens/DishDetailScreen/DishDetailScreen.tsx:9]: Do not use React.FC

File: whizzer/features/menu/screens/RestaurantMenu/RestaurantMenu.tsx [hunk 4]
```diff
@@ -1,11 +1,112 @@
...
+import { supabase } from "../../../../utils/supabase";
+import { Dish } from "../../../../utils/types";
+import CategoriesSlider from "../../components/CategoriesSlider/CategoriesSlider";
+import { FontAwesome } from "@expo/vector-icons";
+import { RootStackScreenProps } from "../../../../types";
+import { RootScreenNames } from "../../../../navigation/constants";
+
+type RestaurantMenuScreenProps = RootStackScreenProps<RootScreenNames.RestaurantMenuScreen>;
+
+const RestaurantMenuScreen: React.FC<RestaurantMenuScreenProps> = ({ navigation }) => {
```
Comment (LRNZ09) [inline whizzer/features/menu/screens/RestaurantMenu/RestaurantMenu.tsx:13]: Do not use React.FC

**Review by LRNZ09 — CHANGES_REQUESTED** [review]

File: whizzer/features/menu/screens/RestaurantMenu/RestaurantMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  navigation,
+  route,
+}) => {
+  const { restaurantId } = route.params;
+
+  const { data, isLoading, error } = useSupabaseHelpers(
+    supabase
+      .from<Dish>("dishes")
+      .select()
+      .filter("restaurant_id", "in", `(${restaurantId})`) // TODO: check is condition
```
Comment (devteapot) [inline whizzer/features/menu/screens/RestaurantMenu/RestaurantMenu.tsx:32]: We could try to filter like this
```suggestion
      .is("restaurant_id", restaurantId)
```

**Review by Fannolo — APPROVED** [review]

**Pull request merged**